### PR TITLE
helpers(product) cleanup, getProductDescription, getProductAttributes

### PR DESCRIPTION
**Pull request by filrak**
- `getProductDescription`
- `getProductAttributes`

### CODE REVIEW

File: packages/commercetools/api-client/src/api/getProduct/defaultQuery.ts [hunk 1]
```diff
@@ -19,11 +19,55 @@ export default gql`
     }
   }
 
+  fragment Attributes on ProductVariant {
+    attributeList {
+            name
+            ... on BooleanAttribute {
```
Comment (andrzejewsky) [inline packages/commercetools/api-client/src/api/getProduct/defaultQuery.ts]: formaring

File: packages/commercetools/helpers/src/index.ts [hunk 2]
```diff
@@ -28,6 +30,23 @@ export const getProductVariants = (products: ProductVariant[], options: any = {}
   return products
 }
 
+export const getProductAttributes = (product: ProductVariant, filterByAttributeName?: Array<string>): any => {
```
Comment (andrzejewsky) [inline packages/commercetools/helpers/src/index.ts]: I guess your intention was `AgnosticProductAttribute` instead of `any`

File: packages/commercetools/helpers/src/types/GraphQL.ts [hunk 3]
```diff
@@ -448,6 +448,8 @@ export type AssetSourceInput = {
 
 export type Attribute = {
   name: Scalars["String"];
+  label?: Maybe<Scalars["String"]>;
```
Comment (andrzejewsky) [inline packages/commercetools/helpers/src/types/GraphQL.ts]: undo these changes as we don't need them

**Review by andrzejewsky — CHANGES_REQUESTED** [review]

File: packages/commercetools/helpers/src/_utils.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    default:
+      return null
+  }
+}
+
+const formatAttributeList = (attributes: any) => {
+  return attributes.map(attr => ({
+    name: attr.name,
+    value: getAttributeValue(attr),
+    label: attr.label ? attr.label : getAttributeValue(attr)
```
Comment (defudef) [inline packages/commercetools/helpers/src/_utils.ts]: `label: attr.label || getAttributeLabel(attr)`

File: packages/commercetools/helpers/src/_utils.ts [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    case 'BooleanAttribute':
+      return attribute.booleanValue
+    case 'ReferenceAttribute': 
+      return { typeId: attribute.typeId, id: attribute.id }
+    default:
+      return null
+  }
+}
+
+const formatAttributeList = (attributes: any) => {
```
Comment (defudef) [inline packages/commercetools/helpers/src/_utils.ts]: what about this approach?
```TS
export const formatAttributeList = (attributes: object[]) => attributes.map(.....)
```

File: packages/commercetools/helpers/src/index.ts [hunk 6]
```diff
@@ -28,6 +30,23 @@ export const getProductVariants = (products: ProductVariant[], options: any = {}
...
+  return (product ? formatAttributeList(product.attributeList) : [])
+  .map(a => ({
+    name: a.name,
+    value: a.value,
+    label: a.label ? a.label : a.value
+  }))
+  .filter(attribute => {
```
Comment (defudef) [inline packages/commercetools/helpers/src/index.ts:35]: Isn't better to use here ternary operator?

```TS
.filter(attribute => filterByAttributeName ? filterByAttributeName.includes(attribute.name) : attribute)
```
```

File: packages/commercetools/helpers/src/index.ts [hunk 7]
```diff
@@ -28,6 +30,23 @@ export const getProductVariants = (products: ProductVariant[], options: any = {}
   return products
 }
 
+export const getProductAttributes = (product: ProductVariant, filterByAttributeName?: Array<string>): any => {
+  return (product ? formatAttributeList(product.attributeList) : [])
+  .map(a => ({
+    name: a.name,
```
Comment (defudef) [inline packages/commercetools/helpers/src/index.ts]: What do you think about this shortcut?
```TS
.map(a => ({
  ...a,
  label: a.label || a.value
}))
```

**Review by defudef — APPROVED** [review]

File: packages/commercetools/helpers/src/_utils.ts (hunk 5 above)
Comment (filrak) [reply packages/commercetools/helpers/src/_utils.ts]: yup, I forgot typings in few places ;) actually there is a type for that

File: packages/commercetools/helpers/src/_utils.ts (hunk 4 above)
Comment (filrak) [reply packages/commercetools/helpers/src/_utils.ts]: I think its equally descriptive :)

File: packages/commercetools/helpers/src/index.ts (hunk 2 above)
Comment (filrak) [reply packages/commercetools/helpers/src/index.ts]: yup

File: packages/commercetools/helpers/src/index.ts (hunk 7 above)
Comment (filrak) [reply packages/commercetools/helpers/src/index.ts]: this code willl copy the unnormalized values like 'stringValue'  so it won't work as intended

File: packages/commercetools/helpers/src/index.ts (hunk 6 above)
Comment (filrak) [reply packages/commercetools/helpers/src/index.ts:35]: imho its equally readable ;)

**Review by andrzejewsky — APPROVED** [review]
